### PR TITLE
Two sgen msbuild properties on the page are for desktop use only

### PR DESCRIPTION
**Pull request by yujayee**
@Lxiamail

### CODE REVIEW

**Comment by PRMerger17** [comment]
@yujayee : Thanks for your contribution! The author, @mikejo5000, has been notified to review your proposed change.

**Comment by Mikejo5000** [comment]
@rainersigwald to verify

**Comment by rainersigwald** [comment]
I'm not sure I understand what the additions mean. Can you expand on it a bit, please?

**Comment by yujayee** [comment]
@rainersigwald The tool sgen.exe can only be used for applications/libraries built on .NET Framework. For .NET core and .NET Standard libraries, users should use Microsoft.XmlSerializer.Generator NuGet package (more info [here](https://github.com/dotnet/corefx/issues/25138)) These two msbuild properties are configurations for sgen.exe, so they should only be used on .NET framework.

**Comment by rainersigwald** [comment]
Ah, I see, thanks!

@Mikejo5000 this gets into a problem with this page. Not all of the properties listed on this page are relevant to every project, and it's hard to tell which is used where. I think this is an improvement, but if you have thoughts on how to make those relationships clearer (even if it starts with a bunch of `?` entries) that could be nice.

**Comment by rainersigwald** [comment]
Actually, you know what's missing here entirely? The global sgen flag. Maybe something like:

| Property or parameter name | Description |
|------------------------------------| - |
| GenerateSerializationAssemblies | A boolean value that indicates whether serialization assemblies should be generated by *SGen.exe*. This property is used for assemblies that target .NET Framework only. To generate serialization assemblies for .NET Standard or .NET Core assemblies, reference the `Microsoft.XmlSerializer.Generator` NuGet package.|

And then cross-reference back to that in these tweak-y properties?

**Comment by PRMerger8** [comment]
@yujayee : Thanks for your contribution! The author, @mikejo5000, has been notified to review your proposed change.

**Comment by yujayee** [comment]
I agree `GenerateSerializationAssemblies `should be included in this page. Updating the pr.

**Comment by yujayee** [comment]
Is the current change looks good? @Mikejo5000

File: docs/msbuild/common-msbuild-project-properties.md [hunk 1]
```diff
@@ -85,8 +85,9 @@ The following table lists frequently used properties that are defined in the Vis
...
-| SGenToolPath | An optional tool path that indicates where to obtain *SGen.exe* when the current version of *SGen.exe* is overridden. |
+| GenerateSerializationAssemblies | Indicates whether XML serialization assemblies should be generated by *SGen.exe*, could be set to on, auto or off. This property is used for assemblies that target .NET Framework only. To generate XML serialization assemblies for .NET Standard or .NET Core assemblies, reference the *Microsoft.XmlSerializer.Generator* NuGet package. |
+| SGenUseProxyTypes | A boolean value that indicates whether proxy types should be generated by *SGen.exe*. Same as *GenerateSerializationAssemblies*, this property is used for .NET Framework only.<br /><br /> The SGen target uses this property to set the UseProxyTypes flag. This property defaults to true, and there is no UI to change this. To generate the serialization assembly for non-webservice types, add this property to the project file and set it to false before importing the *Microsoft.Common.Targets* or the *C#/VB.targets*. |
```
Comment (rainersigwald) [inline docs/msbuild/common-msbuild-project-properties.md]:
```suggestion
| SGenUseProxyTypes | A boolean value that indicates whether proxy types should be generated by *SGen.exe*. This applies only when *GenerateSerializationAssemblies* is true, and for .NET Framework only.<br /><br /> The SGen target uses this property to set the UseProxyTypes flag. This property defaults to true, and there is no UI to change this. To generate the serialization assembly for non-webservice types, add this property to the project file and set it to false before importing the *Microsoft.Common.Targets* or the *C#/VB.targets*. |
```

?

File: docs/msbuild/common-msbuild-project-properties.md [hunk 2]
```diff
@@ -85,8 +85,9 @@ The following table lists frequently used properties that are defined in the Vis
...
 | RemoveIntegerChecks | A boolean value that indicates whether to disable integer overflow error checks. The default value is `false`. This property is equivalent to the `/removeintchecks` switch of the *vbc.exe* compiler. |
-| SGenUseProxyTypes | A boolean value that indicates whether proxy types should be generated by *SGen.exe*.<br /><br /> The SGen target uses this property to set the UseProxyTypes flag. This property defaults to true, and there is no UI to change this. To generate the serialization assembly for non-webservice types, add this property to the project file and set it to false before importing the *Microsoft.Common.Targets* or the *C#/VB.targets*. |
-| SGenToolPath | An optional tool path that indicates where to obtain *SGen.exe* when the current version of *SGen.exe* is overridden. |
+| GenerateSerializationAssemblies | Indicates whether XML serialization assemblies should be generated by *SGen.exe*, could be set to on, auto or off. This property is used for assemblies that target .NET Framework only. To generate XML serialization assemblies for .NET Standard or .NET Core assemblies, reference the *Microsoft.XmlSerializer.Generator* NuGet package. |
```
[the remark this answers is not in the document]
Comment (Mikejo5000) [reply docs/msbuild/common-msbuild-project-properties.md]:
```suggestion
| GenerateSerializationAssemblies | Indicates whether XML serialization assemblies should be generated by *SGen.exe*, which can be set to on, auto, or off. This property is used for assemblies that target .NET Framework only. To generate XML serialization assemblies for .NET Standard or .NET Core assemblies, reference the *Microsoft.XmlSerializer.Generator* NuGet package. |
```

File: docs/msbuild/common-msbuild-project-properties.md [hunk 3]
```diff
@@ -85,8 +85,9 @@ The following table lists frequently used properties that are defined in the Vis
...
+| SGenUseProxyTypes | A boolean value that indicates whether proxy types should be generated by *SGen.exe*. Same as *GenerateSerializationAssemblies*, this property is used for .NET Framework only.<br /><br /> The SGen target uses this property to set the UseProxyTypes flag. This property defaults to true, and there is no UI to change this. To generate the serialization assembly for non-webservice types, add this property to the project file and set it to false before importing the *Microsoft.Common.Targets* or the *C#/VB.targets*. |
+| SGenToolPath | An optional tool path that indicates where to obtain *SGen.exe* when the current version of *SGen.exe* is overridden. Same as *GenerateSerializationAssemblies*, this property is used for .NET Framework only.|
```
Comment (Mikejo5000) [inline docs/msbuild/common-msbuild-project-properties.md]:
```suggestion
| SGenToolPath | An optional tool path that indicates where to obtain *SGen.exe* when the current version of *SGen.exe* is overridden. This property is used for .NET Framework only.|
```

Comment (Mikejo5000) [reply docs/msbuild/common-msbuild-project-properties.md]: I believe this is what is meant here by "Same as *GenerateSerializationAssemblies*?

**Comment by Mikejo5000** [comment]
@yujayee A few minor changes suggested. Thanks!

**Comment by PRMerger14** [comment]
@yujayee : Thanks for your contribution! The author, @mikejo5000, has been notified to review your proposed change.

**Comment by yujayee** [comment]
Thanks guys! I just made the changes according to your feedback.

**Comment by Mikejo5000** [comment]
#sign-off Thanks!

**Pull request merged**